.travis.yml

```diff
@@ -1,66 +1,34 @@
 dist: xenial
-sudo: required
+os: linux
 language: cpp
 
-services:
-  - xvfb
-
-matrix:
-  include:
-    - os: linux
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-6
-      env: COMPILER=gcc GCC=6
-    - os: linux
-      if: branch = master
-      addons:
-        apt:
-          sources:
-            - ubuntu-toolchain-r-test
-          packages:
-            - g++-7
-      env: COMPILER=gcc GCC=7
+stages:
+  - test
+  - singularity-build-push
 
 env:
   global:
     - DISPLAY=":99.0"
-    - MINCONDA_VERSION="latest"
-    - MINCONDA_LINUX="Linux-x86_64"
```

Comment:
I have those OS environment because I will have unittest for macos. I am not sure about windows since it will require a lot of effort on the C++ side. Will it also be easy to do this in your test workflow?

Reply:
I think something like this would be possible by just changing the line at the top from `os: linux` to:

```yml
os:
 - linux
 - osx
```

And then adding something like:

```yml
before_install:
 - if [ "$TRAVIS_OS_NAME" = "linux" ]; then export MINICONDA_OS="Linux-x86_64"  ; fi
 - if [ "$TRAVIS_OS_NAME" = "osx" ]; then export MINICONDA_OS="MacOSX-x86_64"   ; fi
```

That should then lead to  a 2x2 test matrix of tests running on both Linux and MacOSX, for two versions of python (3.7.5 and 3.6.9).

Reply:
I do something like the following in my other projects. 
```
matrix:
    - os: linux
      addons:
        apt:
          sources:
            - ubuntu-toolchain-r-test
          packages:
            - g++-7
    - os: osx
      osx_image: xcode9.3
      compiler: clang
```

And we may not want to increase the number of tests to 2x2 because of Python versions. I think having one test for Python 3.6.9 should be enough and the rest should only use Python 3.7.5.

I raised this question since the matrix is removed. I am just wondering whether it is possible to keep it, or you have better ways to realize all the tests (in and not in containers as well as different compiler/Python versions).